twofactor/queue.go

```diff
@@ -0,0 +1,100 @@
+package twofactor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/enriquebris/goconcurrentqueue"
+	"github.com/vocdoni/saas-backend/internal"
+	"go.vocdoni.io/dvote/log"
+)
+
+type challengeData struct {
+	userID     internal.HexBytes
+	electionID internal.HexBytes
+	contact    string
+	challenge  string
+	startTime  time.Time
+	retries    int
+	success    bool
+}
+
+func (c challengeData) String() string {
+	return fmt.Sprintf("%s[%s]", c.contact, c.challenge)
+}
+
+type Queue struct {
+	queue         *goconcurrentqueue.FIFO
+	ttl           time.Duration
+	throttle      time.Duration
+	sendChallenge []SendChallengeFunc
```

Comment:
With my approach this is no longer needed an should be replaced by the `notification.NotificationService` interface